api/groups/baseGroup.go

```diff
@@ -62,36 +67,53 @@ func (bg *baseGroup) RemoveEndpoint(path string) error {
 	}
 
 	bg.Lock()
-	delete(bg.endpoints, path)
+	for i := 0; i < len(bg.endpoints); i++ {
+		if bg.endpoints[i].Path == path {
+			bg.endpoints = append(bg.endpoints[:i], bg.endpoints[i+1:]...)
```

Comment:
should add a break here as otherwise I do not thing the for loop will work as expected. Please add unit tests.

Reply:
done. there was nothing wrong, just that the slice would have been iterated with no need after the deletion.
also, this situation is caught in the unit test